### PR TITLE
Classification: fixes and improved plugin

### PR DESCRIPTION
**Pull request by sgiraudot**
## Summary of Changes

* Some fixes for the errors/warnings that appeared by testing random forest (using OpenCV) on Windows
* The widget of the plugin has been improved to be much smaller and easier to understand and use

## Release Management

* Affected package(s): Classification / Polyhedron demo

### CODE REVIEW

**Comment by afabri** [comment]
I tested it successfully on Windows.

**Comment by sgiraudot** [comment]
I made a small addition (deleting the classification objets when hiding the widget, especially to clean the color properties of the object that are used by the classification plugin to display labels or feature values).

**Comment by lrineau** [comment]
There is a warning in the polyhedron demo:
```
Plugins/Classification/Classification_plugin.cpp:836:38: warning: comparison of integers of different signs: 'int' and 'size_type' (aka 'unsigned long') [-Wsign-compare]
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-59/Polyhedron_Demo/TestReport_cgaltester_x86-64_Darwin-13.0_Apple-clang-5.0_Release.gz

**Comment by sgiraudot** [comment]
Warning is fixed, but do not merge right now, I still have a fix to do.

**Comment by sgiraudot** [comment]
Ok, it's ready to be tested again.

File: Polyhedron/demo/Polyhedron/Plugins/Classification/Classification_plugin.cpp [hunk 1]
```diff
@@ -835,7 +835,7 @@ public Q_SLOTS:
       if (label_buttons.size() > 1)
         for (int i = position + 1; i < label_buttons.size(); ++ i)
         {
-          int position = i - 1;
+          int position = static_cast<int>(i - 1);
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/Classification/Classification_plugin.cpp]: This does not fix any warning. `i` is already an `int` and the expression `i - 1` is also an `int`.

The warning was about the comparison in the condition of the `for`-loop:
```
/Users/cgaltester/cgal_test/CGAL-4.12-Ic-59/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/test/Polyhedron_Demo/Plugins/Classification/Classification_plugin.cpp:836:38: warning: comparison of integers of different signs: 'int' and 'size_type' (aka 'unsigned long') [-Wsign-compare]
        for (int i = position + 1; i < label_buttons.size(); ++ i)
                                   ~ ^ ~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

Comment (sgiraudot) [reply Polyhedron/demo/Polyhedron/Plugins/Classification/Classification_plugin.cpp]: Damn, you're right… sorry about that, I'll fix it.

**Pull request merged**